[CORE-10097] rptests/services: fix ocsf server startup issue

IoannisRP opened the pull request:
Fixes: [CORE-10097](https://redpandadata.atlassian.net/browse/CORE-10097)

Startup of ocsf server in arm64 seems to be taking slightly longer than 10s (~12 seconds).
Increase startup timeout to 10s -> 30s.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [x] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none


[CORE-10097]: https://redpandadata.atlassian.net/browse/CORE-10097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

IoannisRP commented:
/ci-repeat 1
arch=arm64
tests/rptest/tests/audit_log_test.py

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/67661-->
<!--ff59e64b-d7cd-48c8-83ae-f74e68b8f526-->
<!--9354a7d6-8189-457b-91fa-20b204d2e82d-->
<!--title-->
## Retry command for [Build#67661](https://buildkite.com/redpanda/redpanda/builds/67661)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_admin_oauth
tests/rptest/tests/audit_log_test.py::AuditLogTestOauth.test_kafka_oauth@{"authz_match":"rbac"}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--859b234e-8440-46b2-b998-bb5b48ee73bb-->
<!--TEST RESULTS-->
<!--b05bdb15-5560-43f9-afbc-14bbf982c6fe-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67661">build#67661</a></summary>


| test_class                 | test_method                  | test_arguments          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   |
|:---------------------------|:-----------------------------|:------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestAdminApi       | test_audit_log_metrics       |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 16/21    | upstream reliability is '91.57303370786516'. current run reliability is '76.19047619047619'. drift is 15.38256 and the allowed drift is set to 50. The test should PASS  |
| AuditLogTestAdminApi       | test_config_rejected         |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 19/21    | upstream reliability is '86.5934065934066'. current run reliability is '90.47619047619048'. drift is -3.88278 and the allowed drift is set to 50. The test should PASS   |
| AuditLogTestAdminAuthApi   | test_excluded_principal      |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 18/21    | upstream reliability is '98.56630824372759'. current run reliability is '85.71428571428571'. drift is 12.85202 and the allowed drift is set to 50. The test should PASS  |
| AuditLogTestKafkaApi       | test_audit_topic_protections |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 18/21    | upstream reliability is '88.0'. current run reliability is '85.71428571428571'. drift is 2.28571 and the allowed drift is set to 50. The test should PASS                |
| AuditLogTestKafkaApi       | test_consume                 |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 17/21    | upstream reliability is '95.3405017921147'. current run reliability is '80.95238095238095'. drift is 14.38812 and the allowed drift is set to 50. The test should PASS   |
| AuditLogTestKafkaApi       | test_management              |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 18/21    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS              |
| AuditLogTestKafkaApi       | test_no_auth_enabled         |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 18/21    | upstream reliability is '87.78947368421053'. current run reliability is '85.71428571428571'. drift is 2.07519 and the allowed drift is set to 50. The test should PASS   |
| AuditLogTestKafkaApi       | test_produce                 |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 17/21    | upstream reliability is '94.31438127090301'. current run reliability is '80.95238095238095'. drift is 13.362 and the allowed drift is set to 50. The test should PASS    |
| AuditLogTestKafkaAuthnApi  | test_authn_messages          |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS               |
| AuditLogTestKafkaAuthnApi  | test_excluded_principal      |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 19/21    | upstream reliability is '87.76119402985074'. current run reliability is '90.47619047619048'. drift is -2.715 and the allowed drift is set to 50. The test should PASS    |
| AuditLogTestKafkaAuthnApi  | test_no_audit_user_authn     |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 16/21    | upstream reliability is '97.65886287625418'. current run reliability is '76.19047619047619'. drift is 21.46839 and the allowed drift is set to 50. The test should PASS  |
| AuditLogTestOauth          | test_admin_oauth             |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 9/21     | upstream reliability is '99.63636363636364'. current run reliability is '42.857142857142854'. drift is 56.77922 and the allowed drift is set to 50. The test should FAIL |
| AuditLogTestOauth          | test_kafka_oauth             | {"authz_match": "acl"}  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 16/21    | upstream reliability is '94.50867052023122'. current run reliability is '76.19047619047619'. drift is 18.31819 and the allowed drift is set to 50. The test should PASS  |
| AuditLogTestOauth          | test_kafka_oauth             | {"authz_match": "rbac"} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 7/21     | upstream reliability is '91.30434782608695'. current run reliability is '33.33333333333333'. drift is 57.97101 and the allowed drift is set to 50. The test should FAIL  |
| AuditLogTestSanctionMode   | test_sanctioning_mode        |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS               |
| AuditLogTestSchemaRegistry | test_sr_audit                |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01a-4b66-89d6-2d047e3a5ec4 | FLAKY         | 20/21    | upstream reliability is '97.28813559322033'. current run reliability is '95.23809523809523'. drift is 2.05004 and the allowed drift is set to 50. The test should PASS   |
| AuditLogTestSchemaRegistry | test_sr_audit_bad_authn      |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e01b-4c1c-af80-9c8e3ba82805 | FLAKY         | 17/21    | upstream reliability is '94.64882943143813'. current run reliability is '80.95238095238095'. drift is 13.69645 and the allowed drift is set to 50. The test should PASS  |
| AuditLogTestsAppLifecycle  | test_app_lifecycle           |                         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67661#01978dd1-e018-44c1-ab2a-0348b2d3ed00 | FLAKY         | 19/21    | upstream reliability is '99.63768115942028'. current run reliability is '90.47619047619048'. drift is 9.16149 and the allowed drift is set to 50. The test should PASS   |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/67746">build#67746</a></summary>


| test_class                    | test_method                               | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:------------------------------|:------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeDelayedEnablementTest | test_enabling_iceberg_in_existing_cluster | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1}                                                                                 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67746#01979dab-d965-4dd5-87c9-7880eee0caec | FLAKY         | 15/21    | upstream reliability is '73.47560975609755'. current run reliability is '71.42857142857143'. drift is 2.04704 and the allowed drift is set to 50. The test should PASS |
| IcebergUsageTest              | test_iceberg_usage                        | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "spark"}                                                      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67746#01979db2-7442-4498-81c6-41808c417b80 | FLAKY         | 17/21    | upstream reliability is '86.34020618556701'. current run reliability is '80.95238095238095'. drift is 5.38783 and the allowed drift is set to 50. The test should PASS |
| MaintenanceTest               | test_maintenance_sticky                   | {"use_rpk": true}                                                                                                                      | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67746#01979db2-7440-47e2-a151-939b9102fb6d | FLAKY         | 20/21    | upstream reliability is '97.27722772277228'. current run reliability is '95.23809523809523'. drift is 2.03913 and the allowed drift is set to 50. The test should PASS |
| RandomNodeOperationsTest      | test_node_operations                      | {"cloud_storage_type": 2, "compaction_mode": "adjacent_merge", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/67746#01979dab-d963-4362-a236-49780143b593 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS             |
</details>
<!--/body-->

IoannisRP commented:
/ci-repeat 1
arch=arm64
tests/rptest/tests/audit_log_test.py

IoannisRP committed:
rptests/services: Add timeout parameter in OcsfServer::get_api_version

IoannisRP committed:
rptests/services: Update ocsf start node

- Flatten nested wait_until calls.
- Increase timeout from 10s -> 30s. Arm64 executions seem to be taking
  slightly longer to initialize the server.

IoannisRP commented:
/ci-repeat 1
arch=arm64
tests/rptest/tests/audit_log_test.py

IoannisRP commented:
/ci-repeat 1
arch=arm64
tests/rptest/tests/audit_log_test.py

IoannisRP commented:
/ci-repeat 1
arch=arm64
tests/rptest/tests/audit_log_test.py

michael-redpanda commented:
Thank you for looking into this!

andrewhsu commented:
not exactly sure why the `/ci-repeat` command [failed](https://buildkite.com/redpanda/redpanda/builds/67737/steps/table?jid=01979d5c-e09b-4bd2-888e-b41f7f9b4d05#01979d5c-e09b-4bd2-888e-b41f7f9b4d05/5369-10918) even though previous run was considered [green](https://buildkite.com/redpanda/redpanda/builds/67711/steps/table?jid=01979c3c-ae07-41b3-99ce-69ccfdd3564f#01979c3c-ae07-41b3-99ce-69ccfdd3564f/5778-6329) @gousteris is it suppose to be green even with `exit status 201`?

i'd suggest waiting for build https://buildkite.com/redpanda/redpanda/builds/67746 that was [triggered](https://github.com/redpanda-data/redpanda/pull/26525#event-18277178599) when this PR was marked for review to finish before retrying the `/ci-repeat` command because issuing a new slash command will cancel running tests
